examples/ex28p.cpp

```diff
@@ -0,0 +1,522 @@
+//                       MFEM Example 28 - Parallel Version
+//
+// Compile with: make ex28p
+//
+// Sample runs:  ex28p
+//               ex28p --visit-datafiles
+//               ex28p --order 4
+//
+//               mpirun -np 4 ex28p
+//
+// Description:  Demonstrates a sliding boundary condition in an elasticity
+//               problem. A trapezoid, roughly as pictured below, is pushed
+//               from the right into a rigid notch. Normal displacement is
+//               restricted, but tangential movement is allowed, so the
+//               trapezoid compresses into the notch.
+//
+//                                       /-------+
+//               normal constrained --->/        | <--- boundary force (2)
+//               boundary (4)          /---------+
+//                                          ^
+//                                          |
+//                                normal constrained boundary (1)
+//
+//               This example demonstrates the use of the ConstrainedSolver
+//               framework.
+//
+//               We recommend viewing Example 2 before viewing this example.
+
+#include "mfem.hpp"
+#include <fstream>
+#include <iostream>
+
+using namespace std;
+using namespace mfem;
+
+/** @brief Build a matrix constraining normal components to zero.
+
+    Given a vector space fespace, and the array constrained_att that
+    includes the boundary *attributes* that are constrained to have normal
+    component zero, this returns a SparseMatrix representing the
+    constraints that need to be imposed.
+
+    Each row of the returned matrix corresponds to a node that is
+    constrained. The rows are arranged in (contiguous) blocks corresponding
+    to the actual constraint; in 3D, a one-row constraint means the node
+    is free to move along a plane, a two-row constraint means it is free
+    to move along a line (eg the intersection of two normal-constrained
+    planes), and a three-row constraint is fully constrained (equivalent
+    to MFEM's usual essential boundary conditions).
+
+    The lagrange_rowstarts array is filled in to describe the structure of
+    these constraints, so that constraint k is encoded in rows
+    lagrange_rowstarts[k] to lagrange_rowstarts[k + 1] - 1, inclusive,
```

Comment:
I think using `constraint_rowstart` vs `lagrange_rowstart` would better describe what these operations are doing throughout the code.